src/C-interface/dense/bml_parallel_dense_typed.c

```diff
@@ -64,10 +64,11 @@ void TYPED_FUNC(
     MPI_Datatype dtype[1];
     dtype[0] = MPI_T;
 
-    int blength = A->N * A->N;
+    int blength[1];
+    blength[0] = A->N * A->N;
     MPI_Aint displ[0];
     displ[0] = addr0 - baseaddr;
-    int mpiret = MPI_Type_create_struct(1, &blength, displ, dtype, newtype);
+    int mpiret = MPI_Type_create_struct(1, blength, displ, dtype, newtype);
```

Comment:
The code looks correct though, although I don't understand why it wasn't working before. `count = 1` should work with `int&`.